document-store/src/main/java/org/hypertrace/core/documentstore/expression/impl/RelationExpression.java

```diff
@@ -21,26 +21,26 @@
  */
 @Value
 @AllArgsConstructor(access = AccessLevel.PRIVATE)
-public class RelationalExpression implements FilteringExpression {
+public class RelationExpression implements FilterTypeExpression {
```

Comment:
I think `RelationalExpression` is more easily understandable.
Look up terminology from https://www.geeksforgeeks.org/what-is-an-expression-and-what-are-the-types-of-expressions/

Reply:
Ideally, `FunctionExpression` should have been `ArithmeticExpression` in this terminology. But, we decided to name it `FunctionExpression` to be more generic (so that we can accommodate operations like `SIZE`, `SUBSTRING`, etc.). 
If consistency is the issue, maybe `FunctionExpression` can be renamed as `FunctionalExpression` instead, and `AggregateExpression` to `AggregationExpression` instead.

Reply:
yes, that can be the other way round, either ways we have a consistent naming

Reply:
Yeah. In any case, I prefer keeping `RelationalExpression`, `RelationalOperator`, `LogicalOperator` and `LogicalExpression` undisturbed since these terminology is not new.